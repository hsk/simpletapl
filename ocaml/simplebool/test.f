/* Examples for testing */

lambda x:Bool. x;
lambda x:Bool.lambda x:Bool. x;

 (lambda x:Bool->Bool. if x false then true else false) 
   (lambda x:Bool. if x then false else true); 
a:Bool;
a;
(lambda x:Bool. x) a;
(lambda x:Bool. (lambda x:Bool. x) x) a;
(lambda x:Bool. (lambda x:Bool. x) x) true;
