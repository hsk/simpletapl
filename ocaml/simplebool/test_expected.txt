lambda x:Bool.x : Bool -> Bool
lambda x:Bool.lambda x:Bool.x : Bool -> Bool -> Bool
true : Bool
a : Bool
a : Bool
(lambda x:Bool.x) a : Bool
(lambda x:Bool.(lambda x:Bool.x) x) a : Bool
true : Bool
